app/src/main/java/com/keylesspalace/tusky/util/AttachmentHelper.kt

```diff
@@ -27,10 +28,16 @@ private fun formatDuration(durationInSeconds: Double): String {
 }
 
 fun List<Attachment>.aspectRatios(): List<Double> {
+    val displayMetrics = Resources.getSystem().displayMetrics
```

Comment:
Not sure how to cache this (or if necessary) in this helper method.

On could move "deviceIsHigher" outside - there are two calling locations.

This should maybe be a dependency ("service", singleton)? But that would need to have state...

Reply:
You can get the device orientation from https://developer.android.com/reference/android/content/res/Configuration.html#orientation.

So maybe take that as a parameter to this function, and let the calling function cache it.

Be careful though -- at the moment activities are *not* recreated on orientation changes (e.g., see https://github.com/tuskyapp/Tusky/blob/develop/app/src/main/AndroidManifest.xml#L57) so if this is cached by an activity or fragment it needs to be updated when it changes.

The activities handling config changes is a source of another bug (https://github.com/tuskyapp/Tusky/issues/3217, which I've just extracted in to https://github.com/tuskyapp/Tusky/issues/3398).

Reply:
Hm, I didn't find an easy (maintainable) solution for moving this outward and/or cache.

getSystem() has a measurable time consumption (13 micro seconds for me) but it is only called once new images are displayed (few times per second for very fast scrolling).

So I'd probably leave it be.

Reply:
Instead of `Resources.getSystem().displayMetrics` you could use `Resources.getConfiguration().orientation` -- that more accurately captures the intent of what you're doing.

Either way, passing the value in as a parameter to this function means that you can write unit tests for this function, which feels important.

However...

Here's an alternative to all of this is which is probably better. Instead of this, what if the `minAspect` and `maxAspect` values were stored as named resources, but in resource directories with different qualifiers?

E.g., (https://developer.android.com/guide/topics/resources/providing-resources#QualifierRules):

- `values`
- `values-land` (device is in landscape orientation)
- `values-sw600dp` (shortest screen dimension is at least 600dp)
- `values-sw600dp-land` (shortest screen dimension is at least 600dp *and* device is in landscape orientation)
- etc

Then let the resource lookup system handle finding the best value? The values should still be passed in as parameters (to make testing easy), but this would move this kind of configuration detail out of code and in to resources, where it arguably belongs.

Reply:
Changed to resource values.

That is very slick and could support different configs in future.

For the moment it is still simply "portrait" and "landscape".